test/unit/pusher_test.rb

```diff
@@ -393,4 +404,18 @@ class PusherTest < ActiveSupport::TestCase
 
     teardown { RubygemFs.mock! }
   end
+
+  context "the gem has been signed and not tampered with" do
+    setup do
+      @gem = gem_file("valid_signature-0.0.0.gem")
+      @cutter = Pusher.new(@user, @gem)
+      @cutter.process
+    end
+
+    should "extracts the certificate chain to the version" do
+      assert_not_nil @cutter.version.cert_chain
+    end
```

Comment:
Please also check for `@cutter.code`. Further the gem file you used is invalid now, please use a larger expiry time.
```
Error:
certificate /CN=snakeoil/DC=example/DC=com not valid after 2021-07-08 08:21:01 UTC
```